llvm/lib/CodeGen/AtomicExpandPass.cpp

```diff
@@ -1441,7 +1414,8 @@ Value *AtomicExpand::insertRMWCmpXchgLoop(
   Builder.SetInsertPoint(BB);
   LoadInst *InitLoaded = Builder.CreateLoad(ResultTy, Addr);
   // Atomics require at least natural alignment.
-  InitLoaded->setAlignment(Align(ResultTy->getPrimitiveSizeInBits() / 8));
+  auto TypeSize = F->getParent()->getDataLayout().getTypeSizeInBits(ResultTy);
+  InitLoaded->setAlignment(Align(TypeSize.getFixedSize() / 8));
```

Comment:
What's the reason behind this?

Reply:
That line asserts when used with a pointer type. I've tried to upstream it as https://reviews.llvm.org/D91256